extension/wallet/src/wallet/extract_loan.rs

```diff
@@ -86,10 +85,7 @@ pub async fn extract_loan(
     .map_err(Error::LoanDetails)?;
 
     storage
-        .set_item(
-            "borrower_state",
-            serde_json::to_string(&(borrower, loan_details.clone())).map_err(Error::Serialize)?,
-        )
+        .set_json_item("borrower_state", &(borrower, loan_details.clone()))
```

Comment:
You could avoid this `.clone` if you would change the argument to a reference. Only a reference is needed for serialization.